plugins/inputs/t128_lte/t128_lte.go

```diff
@@ -0,0 +1,384 @@
+	err := json.Unmarshal(content, &lte128Config)
+	if err != nil {
+		return lteDeviceInterface{}, err
+	}
+	var deviceInterfaceContent lteDeviceInterface
+	for _, router := range lte128Config.Config.Authority.Router {
+		for _, node := range router.Node {
+			for _, deviceType := range node.DeviceInterface {
+				if deviceType.Type == "lte" {
+					deviceInterfaceContent = lteDeviceInterface{
+						Name:            deviceType.Name,
+						Type:            deviceType.Type,
+						TargetInterface: deviceType.TargetInterface,
+					}
+					return deviceInterfaceContent, nil
+				}
+			}
+		}
+	}
+	return lteDeviceInterface{}, fmt.Errorf("Cannot find the type")
+}
+
+func (plugin *T128Lte) getLteStateInfo(deviceInterfaceData lteDeviceInterface, pathDir string) (lteDeviceState, error) {
+	var lteState lteDeviceState
+	targetName := deviceInterfaceData.TargetInterface
+	if targetName == "" {
+		return lteDeviceState{}, fmt.Errorf("Cannot find target intferace")
+	}
+	stateFile := filepath.Join(pathDir, fmt.Sprint(targetName, ".state"))
+	_, exist := statFile(stateFile)
+	if !exist {
+		return lteDeviceState{}, fmt.Errorf("Unable to load data from %s", stateFile)
+	}
+	lteStateInfoContent, _ := ioutil.ReadFile(stateFile)
+	err := json.Unmarshal(lteStateInfoContent, &lteState)
+	if err != nil {
+		return lteDeviceState{}, fmt.Errorf("Cannot unmarshal LTE State Info %s : %w", stateFile, err)
+	}
+	return lteState, nil
+}
+
+func (plugin *T128Lte) getLteInfo(deviceInterfaceData lteDeviceInterface, pathDir string) (lteDeviceInfo, error) {
+	var lteInfo lteDeviceInfo
+	targetName := deviceInterfaceData.TargetInterface
+	if targetName == "" {
+		return lteDeviceInfo{}, fmt.Errorf("Cannot find target intferace")
+	}
+	infoFile := filepath.Join(pathDir, fmt.Sprint(targetName, ".info"))
+	_, exist := statFile(infoFile)
+	if !exist {
+		return lteDeviceInfo{}, fmt.Errorf("Unable to load data from %s", infoFile)
+	}
+	lteInfoContent, _ := ioutil.ReadFile(infoFile)
+	err := json.Unmarshal(lteInfoContent, &lteInfo)
+	if err != nil {
+		return lteDeviceInfo{}, fmt.Errorf("Cannot unmarshal LTE State Info %s : %w", lteInfoContent, err)
+	}
+	return lteInfo, nil
+}
+
+func statFile(filePath string) (info fs.FileInfo, exists bool) {
+	info, err := os.Stat(filePath)
+	if err != nil {
+		return nil, false
+	}
+	return info, true
+}
+
+func getServiceStatus(lteMetric lteDeviceState) string {
+	if lteMetric.LTE.RegistrationStatus == "registered" && lteMetric.LTE.ConnectionStatus == "connected" {
+		return "normal"
+	}
+	if lteMetric.LTE.Failure.Reason != "" {
+		registration_status := serviceStatusMap[lteMetric.LTE.Failure.Reason]
+		if registration_status == "" {
+			return lteMetric.LTE.Failure.Reason
+		}
+		return serviceStatusMap[lteMetric.LTE.Failure.Reason]
+	}
+	return lteMetric.LTE.RegistrationStatus
+}
+
+func getSignalValue(lteMetric lteDeviceState) map[string]string {
+	signalStrengths := make(map[string]string)
+	values := reflect.ValueOf(lteMetric.LTE)
```

Comment:
Why is there still reflection in here? We know the fields. Just write out the code to handle them. It will be clearer and more performant.

Reply:
That's fine. If they don't exist, skip them. Reflection is hard to understand and inefficient. Since we're dealing with well defined structs, we shouldn't need it.